addons/mrp_subcontracting/models/stock_picking.py

```diff
@@ -147,17 +147,33 @@ def _prepare_subcontract_mo_vals(self, subcontract_move, bom):
 
     def _subcontracted_produce(self, subcontract_details):
         self.ensure_one()
+
+        group_move = {}
+        group_by_company = defaultdict(list)
         for move, bom in subcontract_details:
             if float_compare(move.product_qty, 0, precision_rounding=move.product_uom.rounding) <= 0:
                 # If a subcontracted amount is decreased, don't create a MO that would be for a negative value.
                 continue
-            mo = self.env['mrp.production'].with_company(move.company_id).create(self._prepare_subcontract_mo_vals(move, bom))
-            self.env['stock.move'].create(mo._get_moves_raw_values())
-            self.env['stock.move'].create(mo._get_moves_finished_values())
-            mo.date_planned_finished = move.date  # Avoid to have the picking late depending of the MO
-            mo.action_confirm()
-
+            mo_subcontract = self._prepare_subcontract_mo_vals(move, bom)
+            # Link the move to the id of the MO's procurement group
+            group_move.setdefault(mo_subcontract['procurement_group_id'], []).append(move)
```

Comment:
You could use a defaultdict here also. It's strange to do both group_by_company that a default dict and group_move using the setdefault. (It's a detail)